bot/cogs/information.py

```diff
@@ -26,6 +27,51 @@ class Information(Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    @staticmethod
+    def role_can_read(channel: GuildChannel, role: Role) -> bool:
+        """Return True if `role` can read messages in `channel`."""
+        overwrites = channel.overwrites_for(role)
+        return overwrites.read_messages is True
+
+    def get_staff_channel_count(self, guild: Guild) -> int:
+        """
+        Get the number of channels that are staff-only.
+
+        We need to know two things about a channel:
+        - Does the @everyone role have explicit read deny permissions?
+        - Do staff roles have explicit read allow permissions?
+
+        If the answer to both of these questions is yes, it's a staff channel.
+        """
+        channel_ids = set()
+        for channel in guild.channels:
+            if channel.type is ChannelType.category:
+                continue
+
+            if channel in channel_ids:
+                continue  # Only one of the roles has to have read permissions, not all
+
+            everyone_can_read = self.role_can_read(channel, guild.default_role)
+
+            for role in constants.STAFF_ROLES:
+                role_can_read = self.role_can_read(channel, guild.get_role(role))
+                if role_can_read and everyone_can_read is False:
```

Comment:
Super nit-pick, but since the variable names read so nicely we could do:
```python
if role_can_ready and not everyone_can_ready:
```
Since they are both bools we don't risk accidentally getting something falsey.

Reply:
You're right. I was concerned that it may be `None`, but since I explicitly use `is` in the function, it should be always be a bool. In hindsight, it's misleading to return a bool cause a value of `None` doesn't indicate one way or another if a role can read.